pandas/io/excel/_base.py

```diff
@@ -667,6 +667,17 @@ class ExcelWriter(metaclass=abc.ABCMeta):
         be parsed by ``fsspec``, e.g., starting "s3://", "gcs://".
 
         .. versionadded:: 1.2.0
+    if_sheet_exists : {'new', 'replace', 'overwrite', 'fail'}, default 'new'
+        How to behave when trying to write to a sheet that already
+        exists (append mode only).
+
+        * new: Create a new sheet, with a name determined by the engine.
+        * replace: Delete the contents of the sheet before writing to it.
```

Comment:
this is too many options, new /replace / error are fine, overwrite is just to specific here and confusing.

Reply:
If the formatting is preserved, then it seems to me this is a commonly desired feature - [SO search](https://stackoverflow.com/search?q=pandas+excel+template&searchOn=3). I've wanted to use this in my own work, where instead I used openpyxl directly and wrote out a frame cell-by-cell.

If formatting is not preserved, then I agree overwrite should not be added.

Reply:
I am also -1 on implementing overwrite

Reply:
@WillAyd - Can I get your thoughts as to why? (Assuming formatting is preserved) To me it seems useful and commonly asked for, and the implementation is two lines.

Reply:
Overwrite versus replace isn't the best semantically. We also don't support overwrite with any other type of IO methods, and to your point above there is a lot of ambiguity as to what "overwrite" should actually overwrite

Reply:
I agree they are close semantically but I think these are the two most useful options and they have fairly different use cases:

Replace: I want to write a table to a "raw data" tab in a spreadsheet with other sheets showing views of that data.

Overwrite: I want to write a table to a specific location in a sheet which contains other stuff - text, graphs, images, other tables, formatting, etc.

I was working in an Excel-heavy area for the last couple of months and I've seen both of these a lot, and they are both well represented in SO questions etc. I think omitting either of them would leave a hole in the API. Do you think there is a way we could make the names and/or docs for these options clearer?

If I was going to suggest an option to get rid of it would be `new`. This is the current behaviour but I'm not sure I can think of a good justification for it, and it can be fairly easily replicated in user code e.g.
```
with ExcelWriter(...) as writer:
    if "sheetname" in writer.book.sheetnames:
        new_sheet_name = new_sheet_names()
    df.to_excel(writer, new_sheet_name)
```

Reply:
We can continue the discussion in a dedicated issue for it but I think trying to do overwrite as part of this PR is going to be a tough sell. It might be better as a separate library, or if it is something we discuss more and want to include in pandas we I think would just want to spend a little bit more time going over the API. Would rather not hold up the other components you have here just for that

Reply:
> We also don't support overwrite with any other type of IO methods

I don't believe other IO methods have the use case of writing to an existing template for formatting purposes, which is what I see as being the primary case here.

> and to your point above there is a lot of ambiguity as to what "overwrite" should actually overwrite

That was not my point, I merely wanted to make sure existing formatting was preserved if the DataFrame was not formatted and indeed that is the case. Not certain what you find unclear.

> or if it is something we discuss more and want to include in pandas we I think would just want to spend a little bit more time going over the API

Other than  what name to call this, what API choices are there?

Reply:
@jreback - would like to get your thoughts here as well, to see if you found my arguments above at all compelling.